cmd/event-received/sirius_event_handler.go

```diff
@@ -333,13 +350,79 @@ func handlePriorityCorrespondenceSent(ctx context.Context, client dynamodbClient
 
 	donor, err := getDonorByLpaUID(ctx, client, v.UID)
 	if err != nil {
-		return fmt.Errorf("failed to get lpa: %w", err)
+		return fmt.Errorf("failed to get donor: %w", err)
 	}
 
 	donor.PriorityCorrespondenceSentAt = v.SentDate
 
 	if err := putDonor(ctx, donor, now, client); err != nil {
-		return fmt.Errorf("failed to update lpa: %w", err)
+		return fmt.Errorf("failed to update donor: %w", err)
+	}
+
+	return nil
+}
+
+type changeConfirmedEvent struct {
+	UID       string     `json:"uid"`
+	ActorType actor.Type `json:"actorType"`
+	ActorUID  string     `json:"actorUID"`
+}
+
+func handleChangeConfirmed(ctx context.Context, client dynamodbClient, event *events.CloudWatchEvent, now func() time.Time, lpaStoreClient LpaStoreClient, materialChange bool) error {
+	var v changeConfirmedEvent
+	if err := json.Unmarshal(event.Detail, &v); err != nil {
+		return fmt.Errorf("failed to unmarshal detail: %w", err)
+	}
+
+	switch v.ActorType {
+	case actor.TypeDonor:
+		donor, err := getDonorByLpaUID(ctx, client, v.UID)
+		if err != nil {
+			return fmt.Errorf("failed to get donor: %w", err)
+		}
+
+		if donor.Tasks.ConfirmYourIdentity.IsPending() && donor.ContinueWithMismatchedIdentity {
+			if materialChange {
+				donor.MaterialChangeConfirmedAt = now()
+				donor.Tasks.ConfirmYourIdentity = task.IdentityStateProblem
+			} else {
+				donor.ImmaterialChangeConfirmedAt = now()
+				donor.Tasks.ConfirmYourIdentity = task.IdentityStateCompleted
+
+				if err := lpaStoreClient.SendDonorConfirmIdentity(ctx, donor); err != nil {
```

Comment:
you'll need to check if the Lpa has been sent to the lpa-store

Reply:
or ignore the 404 error

Reply:
Good point, I'll check for not found.